internal/pkg/manifest/svc.go

```diff
@@ -272,17 +273,49 @@ func (a *AdvancedCount) validScalingFields() []string {
 	}
 }
 
-func (a *AdvancedCount) hasScalingFieldsSet() bool {
-	switch a.workloadType {
-	case LoadBalancedWebServiceType:
-		return !a.CPU.IsEmpty() || !a.Memory.IsEmpty() || !a.Requests.IsEmpty() || !a.ResponseTime.IsEmpty()
-	case BackendServiceType:
-		return !a.CPU.IsEmpty() || !a.Memory.IsEmpty() || !a.Requests.IsEmpty() || !a.ResponseTime.IsEmpty()
-	case WorkerServiceType:
-		return !a.CPU.IsEmpty() || !a.Memory.IsEmpty() || !a.QueueScaling.IsEmpty()
-	default:
-		return !a.CPU.IsEmpty() || !a.Memory.IsEmpty() || !a.Requests.IsEmpty() || !a.ResponseTime.IsEmpty() || !a.QueueScaling.IsEmpty()
+func (a *AdvancedCount) getValidScalingFieldsSet() ([]string, error) {
+	scalingFieldNames := []string{"CPU", "Memory", "Requests", "ResponseTime", "QueueScaling"}
+	validFieldsSet := []string{}
+	invalidFieldsSet := []string{}
+
+	validTags := a.validScalingFields()
+	typ := reflect.ValueOf(*a).Type()
+
+	for _, name := range scalingFieldNames {
+		currStructField, ok := typ.FieldByName(name)
+		if ok {
+			zero := reflect.Zero(currStructField.Type).Interface()
+			value := reflect.ValueOf(a).Elem().FieldByName(name).Interface()
+			if zero != value {
```

Comment:
Similarly to reduce indented lines.
```suggestion
			if zero == value {
			  continue
			}
			...
```

Reply:
I like continuing if the bool value returns not ok in the previous comment, but personally I find it less readable to have a bunch of cases that are for skipping, so in this case I'd prefer to keep nesting the if statement. Lmk if you really disagree with this, otherwise I'll keep it the same for now. 
This is what it looks like currently:
```
for _, name := range scalingFieldNames {
	currStructField, ok := typ.FieldByName(name)
	if !ok {
		continue
	}

	zero := reflect.Zero(currStructField.Type).Interface()
	value := reflect.ValueOf(a).Elem().FieldByName(name).Interface()

	if zero != value {
		tag := currStructField.Tag.Get("yaml")
		isValid := false

		for _, v := range validTags {
			if tag == v {
				validFieldsSet = append(validFieldsSet, tag)
				isValid = true
			}
		}
		if !isValid {
			invalidFieldsSet = append(invalidFieldsSet, tag)
		}
	}
}
```